packages/core/lib/src/core_data.dart

```diff
@@ -12,12 +12,13 @@ part 'data/text_style.dart';
 /// A building element metadata.
 abstract class BuildMetadata {
   /// The associatd element.
-  final dom.Element element;
+  final dom.Element? element;
```

Comment:
We have a lot of `element!` usages which is quite verbose and a bit hard to use. The only place where this can be `null` is the `_rootMeta` so I'm thinking about making this non-nullable:

```suggestion
  final dom.Element element;
```

Then during the `initState` logic make up a pseudo element, something like this:

```dart
_rootMeta = builder.BuildMetadata(dom.Element.tag('root'), _rootTsb);
```

What do you think?

Reply:
Sounds good!